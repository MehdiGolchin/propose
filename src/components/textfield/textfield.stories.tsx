import React from 'react';
import { storiesOf } from '@storybook/react';
import { TextField } from './index';
import { ThemeProvider } from 'styled-components';
import { Default } from '../../themes';

storiesOf('TextField', module)
  .add('Text', () => (
    <ThemeProvider theme={Default}>
      <TextField title="Text" />
    </ThemeProvider>
  ))
  .add('Number', () => (
    <ThemeProvider theme={Default}>
      <TextField title="Number" type="number" />
    </ThemeProvider>
  ))
  .add('Password', () => (
    <ThemeProvider theme={Default}>
      <TextField title="Password" type="password" />
    </ThemeProvider>
  ));
