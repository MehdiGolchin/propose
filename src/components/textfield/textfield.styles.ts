import styled, { css } from 'styled-components';

export const Label = styled.label`
  ${({ theme }) => css`
    font-family: ${theme.fontFamily};
    color: ${theme.fontColor};
    font-size: ${theme.fontSize};
  `}
`;

export const Input = styled.input`
  display: block;
  line-height: 30px;
  ${({ theme }) => css`
    font-family: ${theme.fontFamily};
    color: ${theme.fontColor};
    font-size: ${theme.fontSize};
  `}
`;
