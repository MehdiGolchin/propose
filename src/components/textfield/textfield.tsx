import React, { FunctionComponent, useCallback, useState } from 'react';
import { Label, Input } from './textfield.styles';

export type TextFieldChange = (newValue: string) => void;

export interface TextFieldProps {
    readonly title: string;
    readonly initialValue?: string;
    readonly type?: 'text' | 'number' | 'password';
    readonly onChange?: TextFieldChange;
}

export const TextField: FunctionComponent<TextFieldProps> = ({ title, initialValue = '', type = 'text', onChange }) => {
    const [value, setValue] = useState(initialValue);

    const onChangeCallback = useCallback((e) => {
        setValue(e.target.Value);
        onChange && onChange(value);
    }, [value]);

    return (
        <Label>{title}
            <Input
                type={type}
                value={value}
                onChange={onChangeCallback}
            />
        </Label>
    );
};