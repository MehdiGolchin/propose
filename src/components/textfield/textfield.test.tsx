import React from 'react';
import { shallow } from 'enzyme';
import { TextField } from './textfield';

describe('TextField', () => {

    it('should match text snapshot', () => {
        const wrapper = shallow(<TextField title="Text" />);
        expect(wrapper).toMatchSnapshot();
    });

    it('should match number snapshot', () => {
        const wrapper = shallow(<TextField title="Text" type="number" />);
        expect(wrapper).toMatchSnapshot();
    });

    it('should match password snapshot', () => {
        const wrapper = shallow(<TextField title="Text" type="password" />);
        expect(wrapper).toMatchSnapshot();
    });

});