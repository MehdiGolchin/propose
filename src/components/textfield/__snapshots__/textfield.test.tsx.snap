// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField should match number snapshot 1`] = `
<styled.label>
  Text
  <styled.input
    type="number"
  />
</styled.label>
`;

exports[`TextField should match password snapshot 1`] = `
<styled.label>
  Text
  <styled.input
    type="password"
  />
</styled.label>
`;

exports[`TextField should match text snapshot 1`] = `
<styled.label>
  Text
  <styled.input
    type="text"
  />
</styled.label>
`;
